tests/src/test/scala/whisk/core/controller/test/EntitlementProviderTests.scala

```diff
@@ -681,4 +689,25 @@ class EntitlementProviderTests extends ControllerTestCommon with ScalaFutures {
         Await.ready(check, requestTimeout).eitherValue.get shouldBe expected
     }
   }
+
+  it should "restrict access to disallowed action kinds for a subject" in {
+    implicit val tid = transid()
+    implicit val ep = entitlementProvider
+    val subject = WhiskAuthHelpers.newIdentity().copy(limits = UserLimits(allowedKinds = Some(allowedKinds)))
+
+    disallowedKinds.foreach(k =>
+      intercept[RejectRequest] {
+        Await.result(ep.check(subject, Some(getExec(k))), 1.seconds)
```

Comment:
Can you add a check on the reject request error message as well?

Reply:
Good catch. I've added the message checker